mutation/mutation_partition_v2.cc

```diff
@@ -243,6 +249,15 @@ stop_iteration mutation_partition_v2::apply_monotonically(const schema& s, const
         }
     });
 
+    // This guard will ensure that LSA reserves one free segment more than it
+    // needs for internal reasons.
+    //
+    // It will be destroyed immediately before the sentinel-inserting `defer`
+    // happens, ensuring that the sentinel insertion has at least one free LSA segment
+    // to work with. This should be enough, since we only need to allocate a few
+    // B-tree nodes.
+    auto memory_reserve_for_sentinel_inserts = hold_reserve(logalloc::segment_size);
```

Comment:
To guarantee this ordering stays the same, you can move the guard into the capture list of the lambda passed to `defer()`, then release it manually in the first line. That is more explicit than relying on destruction order, which can be changed inadvertently by a later refactoring. Maybe I'm just overly paranoid, I'll leave it up to you.

Reply:
Too late, Tomek queued the patch before I could respond.

Being more explicit here would probably be a good idea here, because the order is indeed very non-obvious here.

But it probably won't matter, simply `mutation_partition_v2::apply_monotonically()` is something that nobody should touch with a ten-meter pole before studying the details for 2 weeks. One does not simply "refactor" it.

The refactorer would have to just ignore the comments to mess up this particular ordering.